sdk/trace/provider_test.go

```diff
@@ -94,3 +99,151 @@ func TestSchemaURL(t *testing.T) {
 	tracerStruct := tracerIface.(*tracer)
 	assert.EqualValues(t, schemaURL, tracerStruct.instrumentationLibrary.SchemaURL)
 }
+
+func TestTracerProviderSamplerConfigFromEnv(t *testing.T) {
+	type testCase struct {
+		sampler            string
+		samplerArg         string
+		argOptional        bool
+		description        string
+		errorStr           string
+		invalidArgErrorStr string
```

Comment:
This will cause these tests to be brittle.  If you want to test a particular kind of error happen we need to make sentinel errors, and check with `error.Is`

Reply:
Fixed, should be a little more idiomatic now.